wire-ios-data-model/Source/MLS/CoreCryptoKeyProvider.swift

```diff
@@ -39,25 +45,66 @@ public class CoreCryptoKeyProvider {
             return key
         }
     }
+
+    private func removeLegacyKeyIfNeeded() {
+        let legacyItem = LegacyCoreCryptoKeychainItem()
+
+        do {
+            _ = try KeychainManager.fetchItem(legacyItem) as Data
+            WireLogger.coreCrypto.info("Found legacy core crypto key. Deleting...")
+            try KeychainManager.deleteItem(legacyItem)
+            WireLogger.coreCrypto.info("Deleted legacy core crypto key")
+        } catch KeychainManager.Error.failedToDeleteItemFromKeychain(let error) {
+            WireLogger.coreCrypto.error("Failed to delete legacy core crypto key: \(String(describing: error))")
+        } catch {
+            // key was not found. no action needed
+        }
+    }
+}
+
+class CoreCryptoKeychainItem: KeychainItemProtocol {
+
+    static let tag = "com.wire.mls.key".data(using: .utf8)!
```

Comment:
So we don't have to change the tag?

Reply:
It's not necessary because the legacy vs the new key can be differentiated with the `kSecAttrAccessible` attribute